ada_2_fhir-r4/zibs2020/payload/0.0/nl-core-Alert.xsl

```diff
@@ -0,0 +1,181 @@
+                            <xsl:call-template name="makeReference">
+                                <xsl:with-param name="in" select="."/>
+                                <xsl:with-param name="profile" select="'nl-core-Problem'"/>
+                            </xsl:call-template>
+                        </valueReference>
+                    </extension>
+                </xsl:for-each>
+                
+                <xsl:for-each select="toelichting">
+                    <extension url="http://nictiz.nl/fhir/StructureDefinition/ext-Comment">
+                        <valueString>
+                            <xsl:call-template name="string-to-string">
+                                <xsl:with-param name="in" select="."/>
+                            </xsl:call-template>
+                        </valueString>
+                    </extension>
+                </xsl:for-each>
+                
+                <status>
+                    <xsl:choose>
+                        <xsl:when test="xs:date(eind_datum_tijd/@value) lt current-date() or xs:dateTime(eind_datum_tijd/@value) lt current-dateTime()">
+                            <xsl:attribute name="value" select="'inactive'"/>
+                        </xsl:when>
+                        <xsl:otherwise>
+                            <xsl:attribute name="value" select="'active'"/>
+                        </xsl:otherwise>
+                    </xsl:choose>
+                </status>
+                
+                <xsl:for-each select="alert_type">
+                    <category>
+                        <xsl:call-template name="code-to-CodeableConcept"/>
+                    </category>
+                </xsl:for-each>
+                
+                <xsl:choose>
+                    <xsl:when test="alert_naam">
+                        <xsl:for-each select="alert_naam">
+                            <code>
+                                <xsl:call-template name="code-to-CodeableConcept"/>
+                            </code>
+                        </xsl:for-each>
+                    </xsl:when>
+                    <xsl:when test="not(alert_naam) and conditie/probleem">
+                        <code>
+                            <coding>
+                                <system value="http://terminology.hl7.org/CodeSystem/data-absent-reason"/>
+                                <code value="not-applicable"/>
+                                <display value="Not Applicable"/>
+                            </coding>
+                        </code>
+                    </xsl:when>
+                    <xsl:when test="(alert_naam and conditie/probleem) or (not(alert_naam) and not(conditie/probleem))">
+                        <xsl:message terminate="no">
+                            Warning: exactly one of the following concept Condition::Concern and AlertName SHALL be present.
+                        </xsl:message>
+                    </xsl:when>
```

Comment:
Hier gaat hij alleen aankomen als `(not(alert_naam) and not(conditie/probleem))`. Het eerste statement gaat hij nooit pakken, want als hij een `alert_naam` heeft, slaat hij al aan bij de eerste `xsl:when`.

Aan de andere kant: we hebben soort van afgesproken dat we niet gaan valideren in de mappings, dus misschien kan deze net zo goed weg.

Reply:
Ok! Dat lijkt me opzich ook beter. De externe validatoren en profielen kunnen dat idd beter doen. Ik haal het weg.